### PR TITLE
Utilize new golang 1.8 features within the codebase -#276

### PR DESCRIPTION
**Pull request by flaurida**
Issue #276 - Use `sort.Slice` function rather than re-creating `sort.Sort` interfaces. Use `binary.Read/Write` with `bool` arguments within `channeldb`.

### CODE REVIEW

**Comment by flaurida** [comment]
See that checks are failing, I will look into it. Thanks for your patience :)

**Comment by flaurida** [comment]
Note that I did not delete distanceHeap interface because I saw it was used in findPath function (`pathfind.go`), but am still not sure if I actually could delete some/all of it.

**Comment by flaurida** [comment]
Thanks so much for the review! I appreciate the guidance and will make the suggested fixes within the next 1-2 days. Really excited to be getting up the curve on contributing to LND :)

**Comment by flaurida** [comment]
I tried to find the other instances where `binary.Read/Write` could be used in both `channeldb/channel.go` as well as the `channeldb` package in general. Note that the last two commits are other places in the `channeldb` package where I found `[1]byte`, but I can remove those commits if you prefer. Also tried to fix commit messages and ensure none of my lines were too long.

Thanks again for taking a look and for your patience :)

File: channeldb/invoices.go [hunk 1]
```diff
@@ -329,7 +330,7 @@ func serializeInvoice(w io.Writer, i *Invoice) error {
 	if i.Terms.Settled {
 		settleByte[0] = 1
 	}
-	if _, err := w.Write(settleByte[:]); err != nil {
+	if err := binary.Write(w, byteOrder, settleByte[:]); err != nil {
```
Comment (Roasbeef) [inline channeldb/invoices.go]: We should be able to pass `i.Terms.Settled` directly into the `binary.Write` here, eliminating the declaration of `settleByte` all together.

File: channeldb/invoices.go [hunk 2]
```diff
@@ -379,7 +380,7 @@ func deserializeInvoice(r io.Reader) (*Invoice, error) {
 	invoice.Terms.Value = lnwire.MilliSatoshi(byteOrder.Uint64(scratch[:]))
 
 	var settleByte [1]byte
-	if _, err := io.ReadFull(r, settleByte[:]); err != nil {
+	if err := binary.Read(r, byteOrder, settleByte[:]); err != nil {
```
Comment (Roasbeef) [inline channeldb/invoices.go]: We should be able to pass `i.Terms.Settled` directly into the `binary.Write` here, eliminating the declaration of `settleByte` all together.

File: channeldb/waitingproof.go [hunk 3]
```diff
@@ -212,7 +212,7 @@ func (p *WaitingProof) Encode(w io.Writer) error {
 		b[0] = 1
 	}
 
-	if _, err := w.Write(b[:]); err != nil {
+	if err := binary.Write(w, byteOrder, b[:]); err != nil {
```
Comment (Roasbeef) [inline channeldb/waitingproof.go]: We should be able to pass `p.isRemote` directly into `binary.Write` here, eliminating the declaration of `var b [1]byte` all together.

File: channeldb/waitingproof.go [hunk 4]
```diff
@@ -223,11 +223,11 @@ func (p *WaitingProof) Encode(w io.Writer) error {
 	return nil
 }
 
-// Decode reads the data from the byte stream and initialize the
+// Decode reads the data from the byte stream and initializes the
 // waiting proof object with it.
 func (p *WaitingProof) Decode(r io.Reader) error {
 	var b [1]byte
-	if _, err := r.Read(b[:]); err != nil {
+	if err := binary.Read(r, byteOrder, b[:]); err != nil {
```
Comment (Roasbeef) [inline channeldb/waitingproof.go]: We should be able to pass `&p.isRemote` directly into `binary.Read` here, eliminating the declaration of `var b [1]byte` all together.

**Comment by Roasbeef** [comment]
Once those two minor comments are addressed, I'll get this merged!

**Pull request merged**